third_party/terraform/data_sources/data_source_google_kms_secret.go

```diff
@@ -34,33 +43,34 @@ func dataSourceGoogleKmsSecret() *schema.Resource {
 func dataSourceGoogleKmsSecretRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	cryptoKeyId, err := parseKmsCryptoKeyId(d.Get("crypto_key").(string), config)
-
+	tfCryptoKeyId, err := parseKmsCryptoKeyId(d.Get("crypto_key").(string), config)
 	if err != nil {
 		return err
 	}
 
-	ciphertext := d.Get("ciphertext").(string)
-
-	kmsDecryptRequest := &cloudkms.DecryptRequest{
-		Ciphertext: ciphertext,
+	ciphertext, err := base64.StdEncoding.DecodeString(d.Get("ciphertext").(string))
```

Comment:
So if I'm correct, the new API expects plaintext but the old one expected it to be base64'd?

Reply:
No, but I definitely see how you would think that 😉. The old client library expected `Ciphertext` to be a base64-encoded string. The user was supplying us a b64-encoded string, so we passed it along directly.

The new client library expects a `[]byte`, so we have to decode it first before passing it along to the new library as ciphertext.